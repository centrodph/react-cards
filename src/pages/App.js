import React, { Component } from 'react';
import { renderRoutes } from 'react-router-config';

import Header from '../components/Header';

class App extends Component {
	render() {
		const { route } = this.props;

		return (
			<div className="component-app">
				<Header />
				{renderRoutes(route.routes)}
			</div>
		);
	}
}

export default App;
