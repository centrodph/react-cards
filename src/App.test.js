import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow } from 'enzyme';

import App from './App';

test('Render App', () => {
	const component = renderer.create(<App />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});

describe('Render app', () => {
	it('renders three <Foo /> components', () => {
		const component = renderer.create(<App />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders an `.App-title`', () => {
		const wrapper = shallow(<App />);
		chaiExpect(wrapper.find('.App-title')).to.have.length(1);
	});
});
