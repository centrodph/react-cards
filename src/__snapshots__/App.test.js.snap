// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App 1`] = `
<div
  className="App"
>
  <h1
    className="App-title"
  >
    Welcome to React
  </h1>
</div>
`;

exports[`Render app renders three <Foo /> components 1`] = `
<div
  className="App"
>
  <h1
    className="App-title"
  >
    Welcome to React
  </h1>
</div>
`;
